src/cljs/main/broadfcui/page/workspace/monitor/common.cljs

```diff
@@ -81,27 +81,47 @@
           :data-test-id "status-icon" :data-test-value "unknown"}
    [icons/UnknownIcon {:size 12}]])
 
+(defn sort-order-sub-status [wf-statuses]
+  (cond
+    (contains? wf-statuses :Failed) 20
+    (contains? wf-statuses :Aborted) 30
+    (contains? wf-statuses :Succeeded) 40
+    :else (do (utils/log "Unknown submission status: " wf-statuses) 50)))
 
-(defn icon-for-wf-status [status]
+(defn sort-order-wf-status [status]
   (cond
-    (contains? wf-success-statuses status) (render-success-icon)
-    (contains? wf-running-statuses status) (render-running-icon)
-    (contains? wf-failure-statuses status) (render-failure-icon)
-    :else (do (utils/log "Unknown workflow status: " status)
-              (render-unknown-icon))))
+    (contains? wf-running-statuses status) 60
+    (contains? wf-failure-statuses status) 70
+    (contains? wf-success-statuses status) 80
+    :else (do (utils/log "Unknown workflow status: " status) 90)))
+
+(defn sort-order-submission [sub-status wf-statuses]
+  (if (= "Done" sub-status)
+    (sort-order-sub-status wf-statuses)
+    (sort-order-wf-status sub-status)))
+
+(defn- icon-for-sort-index [sort-index]
+  (case sort-index
+    20 (render-failure-icon)
+    30 (render-failure-icon)
+    40 (render-success-icon)
+    60 (render-running-icon)
+    70 (render-failure-icon)
+    50 (render-unknown-icon)
+    80 (render-success-icon)
+    90 (render-unknown-icon)
+    (render-unknown-icon)))
```

Comment:
in order to understand the code and do a review, I wrote out comments for this block:
```
(defn- icon-for-sort-index [sort-index]
  (case sort-index           ; icon / submission status / workflow status
    20 (render-failure-icon) ; ! / Done / at least one Failed
    30 (render-failure-icon) ; ! / Done / at least one Aborted
    40 (render-success-icon) ; ✓ / Done / at least one Succeeded
    50 (render-unknown-icon) ; ? / unknown submission status / any
    60 (render-running-icon) ; = / Running | Submitted | Queued | Launching / any
    70 (render-failure-icon) ; ! / Failed | Aborting | Aborted / any
    80 (render-success-icon) ; ✓ / Succeeded / any
    90 (render-unknown-icon) ; ? / any except Done / no workflows, or all workflows are unknown
    (render-unknown-icon)))  ; ? / something went wrong in the code
```
is this correct? If so, include or abandon at your discretion!

Reply:
it's close to correct! i think it's a good thing to have, will adjust and keep.